theseus/geometry/lie_group.py

```diff
@@ -44,6 +45,17 @@ def _init_data(*args: Any) -> torch.Tensor:
     def dof(self) -> int:
         pass
 
+    @staticmethod
+    @abc.abstractmethod
+    def rand(
```

Comment:
In `torch` and `numpy`, function `rand()` has the specific meaning of sampling in the range [0, 1]. Wondering if naming this function `rand()` might be confusing to users. Should we use a different name? Maybe `random()`, `sample()`, something else?

Reply:
I don't think this is a big issue. But I do consider using `LieGroup.rand()` and `LieGroup.randn()`  for uniform and for normal distributions, respectively, which is consistent with `torch.rand()` and `torch.randn()`. Any thoughts?

Reply:
Yeah, I actually like that suggestion of having both versions. But in that case, let's make the point types sampling consistent with what the user would expect from the torch. For example, `Point2.rand()` returns points in [0, 1].

Reply:
> Yeah, I actually like that suggestion of having both versions. But in that case, let's make the point types sampling consistent with what the user would expect from `torch`. For example, `Point2.rand()` returns points in [0, 1].

Shall we keep the range of `theta` as `[-pi, pi]`?

Reply:
Yes, I meant that only for subclasses of `Vector`.